### PR TITLE
Changes to include generic2 protocol for sensor node communications.

### PR DESCRIPTION
**Pull request by amoghasgekar**
Designing a new protocol for ATTiny45 based generic sensor node.

### CODE REVIEW

**Comment by sweetpi** [comment]
Nice. Would be cool if you would publish the ATTiny code.

I've added your protocol with some renames to match the other protocols:

 * `battery_level` => `battery`
 * `node_type` => `node`

https://github.com/pimatic/rfcontroljs/blob/master/src/protocols/generic2.coffee#L11-L22

Released with rfcontroljs 0.0.39

File: src/helper.coffee [hunk 1]
```diff
@@ -121,8 +121,16 @@ module.exports = {
     else
       paritybitRef = ((parity % 2) isnt 0)
     return paritybitRef
+	
+  hexChecksum: (data) ->
+    checksum = 0
+    number = @binaryToNumberLSBMSB(data, 0, 31)
```
Comment (Icesory) [inline src/helper.coffee]: This is not used. I think you can remove it or?

Comment (sweetpi) [inline src/helper.coffee]: Whats with https://github.com/pimatic/rfcontroljs/blob/master/src/protocols/generic2.coffee#L37 ?

**Comment by Icesory** [comment]
i mean exact this line.
```coffeescript
number = @binaryToNumberLSBMSB(data, 0, 31)
```
Number isnt used anymore

File: src/helper.coffee (hunk 1 above)
Comment (Icesory) [inline src/helper.coffee]: i mean exact this line.

number = @binaryToNumberLSBMSB(data, 0, 31)
Number isnt used anymore

**Comment by amoghasgekar** [comment]
Thanks for the comments folks! This is my first commit on github, very excited :)
@sweetpi Will publish the ATTiny code too. The goal is to build a very cheap, reliable and long-lived generic sensor node, that can run off a CR2032 cell for roughly 2-3 years without issues.

I couldn't find a way to reply to the comments. Is that because the changes were already merged into the main branch?

**Comment by sweetpi** [comment]
To replay, you have to click on closed, at the pull request list.

File: src/helper.coffee (hunk 1 above)
Comment (sweetpi) [inline src/helper.coffee]: @Icesory ah, all right. I was blind ;)

**Pull request merged**